packages/components/src/templates/next/components/internal/Link/utils/generateAriaLabel.ts

```diff
@@ -0,0 +1,41 @@
+import type { LinkProps } from "~/interfaces/internal/Link"
+
+type GenerateAriaLabelProps = Pick<LinkProps, "isExternal" | "label"> & {
+  textContent?: string
+}
+
+const getDomainFromUrl = (url: string) => {
+  if (!url) return ""
+
+  const urlObj = new URL(url)
+  return urlObj.hostname
+}
+
+const isUrl = (textContent: string) => {
+  if (!textContent) return false
+
+  try {
+    new URL(textContent)
+    return true
+  } catch {
+    return false
+  }
+}
+
+export const generateAriaLabel = ({
+  label,
+  textContent,
+  isExternal = false,
+}: GenerateAriaLabelProps): string | undefined => {
+  if (label) return label
+
+  if (!textContent) return undefined
```

Comment:
[nit] no actionable, since this is parity but i was wondering if just defaulting to `This is a link to an unknown url` would make sense or would it cause more confusion

my guess is that to screen readers (and my assumption is that the MOP here is visually handicapped), at least they know it's a link (but this would alr be done by the html tag) but hte value add seems marginal so ok eitherways

Reply:
text content being falsey is a very edge case e.g. hyperlinking empty spaces (TBH i don't even know if that's possible?). So we can just return undefined and have it not set the aria label

To share more:
- [this hyperlink's text is not a URL but link to google](https://www.google.com) -> does not trigger Oobee must-fix error
- [www.google.com](https://www.google.com) -> this trigger Oobee must-fix error

Reply:
sgtm